domain/agentbinary/service/store.go

```diff
@@ -24,14 +24,14 @@ import (
 
 // State describes the interface that the cache state must implement.
 type State interface {
-	// Add adds a new agent binary's metadata to the database.
+	// AddAgentBinary adds a new agent binary's metadata to the database.
 	// [agentbinaryerrors.AlreadyExists] when the provided agent binary already
 	// exists.
 	// [agentbinaryerrors.ObjectNotFound] when no object exists that matches
 	// this agent binary.
 	// [coreerrors.NotSupported] if the architecture is not supported by the
 	// state layer.
-	Add(ctx context.Context, metadata agentbinary.Metadata) error
+	AddAgentBinary(ctx context.Context, arg agentbinary.AddAgentBinaryArg) error
```

Comment:
Up to you on this one.

Do we still like this as a name? `Add` to me implies that we are doing the physical add of the binary data as well. Would `RegisterAgentBinary` be better or something to that effect?

Up to you if you want to change.